src/dawn/engine_dawn.cc

```diff
@@ -14,15 +14,143 @@
 
 #include "src/dawn/engine_dawn.h"
 
+#include <algorithm>
 #include <utility>
 #include <vector>
 
 #include "dawn/dawncpp.h"
 #include "src/dawn/device_metal.h"
+#include "src/sleep.h"
 
 namespace amber {
 namespace dawn {
 
+namespace {
+
+// The dimensions of the framebuffer.  The code also assumes the framebuffer
+// is 2D and does not have any array layers.
+enum { kFramebufferWidth = 250, kFramebufferHeight = 250 };
```

Comment:
It's a tic I picked up some time ago.  Enum-constants are a bit more pure since they don't allocate storage.

      const uint32_t foo;  //  allocates storage

If it's too weird I can switch.